lib/rucio/daemons/hermes/hermes.py

```diff
@@ -134,12 +135,11 @@ def deliver_emails(once=False, send_email=True, thread=0, bulk=1000, delay=10):
         if once:
             break
 
-        t_delay = delay - (time.time() - t_start)
-        t_delay = t_delay if t_delay > 0 else 0
-        if t_delay:
-            logging.debug('[email] %i:%i - sleeping %s seconds',
-                          heartbeat['assign_thread'], heartbeat['nr_threads'], t_delay)
-        time.sleep(t_delay)
+        if len(messages) < bulk:
+            t_delay = delay - (time.time() - t_start)
```

Comment:
For this one, can't you use the `daemon_sleep` method you wrote for the other PR?

Reply:
Ups, I didn't realise, I'm going to change it